### ISSUE TITLE
Sometimes combination of linting fixes with --safe-auto-correct cause syntax errors

### ISSUE THREAD

**Issue by moger777** [rubocop-hq/rubocop#7667]
## Expected behavior

Running `rubocop --safe-auto-correct` shouldn't cause any syntax issues.

## Actual behavior

It is possible to cause syntax issues with combinations of auto correct, even when using `--safe-auto-correct`

## Steps to reproduce the problem

given the following config in .rubocop.yml:

```yml
# all other rules inherit from default
Style/TrailingCommaInHashLiteral:
  EnforcedStyleForMultiline: consistent_comma
```

and the following ruby code:

```ruby
def print_args(*args)
  puts(args)
end

print_args 1, 2, {
  dar: 'car',
}
```

Running `rubocop --safe-auto-correct` results in the following code with a syntax error:

```ruby
def print_args(*args)
  puts(args)
end
print_args 1, 2, 
  dar: 'car',
```

## RuboCop version

0.79.0

**Comment by koic** [comment]
Thanks for the feedback. `Style/BracesAroundHashParameters` cop will be removed in the next RuboCop release.
https://github.com/rubocop-hq/rubocop/issues/7641

Please running with `Style/BracesAroundHashParameters` disabled.

```yaml
Style/BracesAroundHashParameters:
  Enabled: false
```

Thank you.
